grai-server/app/connections/task_helpers.py

```diff
@@ -48,27 +47,32 @@ def update(
             spec = active_model.__dict__
 
             if type == "Edge":
-                spec["source"] = NodeID(**active_model.source.__dict__)
-                spec["destination"] = NodeID(**active_model.destination.__dict__)
+                spec["source"] = NodeNamedID(**active_model.source.__dict__)
+                spec["destination"] = NodeNamedID(**active_model.destination.__dict__)
+                spec["metadata"] = {
+                    "grai": {"node_type": "Edge", "edge_type": "ColumnToColumn"}
+                }
+            else:
+                spec["metadata"] = {"grai": {"node_type": "Column"}}
```

Comment:
Not at all convinced by these lines, could you take a look @ieaves

Reply:
Not using the schemas ended up being potentially just as bad but so it goes.